src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/README.md

```diff
@@ -0,0 +1,209 @@
+#### ---#if (IsAzureOpenAI)
+## Using Azure OpenAI
+
+To use Azure OpenAI, you will need an Azure account and an Azure OpenAI Service resource. For detailed instructions, see the [Azure OpenAI Service documentation](https://learn.microsoft.com/azure/ai-services/openai/how-to/create-resource).
+
+### 1. Create an Azure OpenAI Service Resource
+Follow the instructions in the [Azure portal](https://portal.azure.com/) to create an Azure OpenAI Service resource.
+
+### 2. Deploy the Models
+Deploy the `gpt-4o-mini` and `text-embedding-3-small` models to your Azure OpenAI Service resource. When creating those deployments, give them the same names as the models (`gpt-40-mini` and `text-embedding-3-small`). See the Azure OpenAI documentation to learn how to [Deploy a model](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/create-resource?pivots=web-portal#deploy-a-model).
+
+#### ---#if (UseManagedIdentity)
+### 3. Configure Azure OpenAI for Keyless Authentication
+This template is configured to use keyless authentication (also known as Managed Identity, with Entra ID). In the Azure Portal, when viewing the Azure OpenAI resource you just created, view access control settings and assign yourself the `Cognitive Services OpenAI User` role. [Learn more](https://learn.microsoft.com/azure/developer/ai/keyless-connections).
+
+### 4. Configure Azure OpenAI Endpoint
+Configure your Azure OpenAI endpoint for this project, using .NET User Secrets:
+   1.  In the Azure Portal, navigate to your Azure OpenAI resource.
+   2.  Copy the "Endpoint" URL from the "Keys & Endpoint" section.
+#### ---#if (hostIdentifier == "vs")
+   3.  In Visual Studio, right-click on your project in the Solution Explorer and select "Manage User Secrets".
+   4.  This will open a `secrets.json` file where you can store your Azure OpenAI endpoint securely. Add the following key & value to the file:
+
+   ```json
+   {
+   "AzureOpenAI:Endpoint": "YOUR-AZURE-OPENAI-ENDPOINT"
+   }
+   ```
+#### ---#else
+   3.  From the command line, configure your Azure OpenAI endpoint for this project using .NET User Secrets by running the following commands:
+
+   ```sh
+   cd <<your-project-directory>>
+   dotnet user-secrets set AzureOpenAI:Endpoint YOUR-AZURE-OPENAI-ENDPOINT
+   ```
+#### ---#endif
+
+   Make sure to replace `YOUR-AZURE-OPENAI-ENDPOINT` with your actual Azure OpenAI endpoint.
+#### ---#else
+### 3. Configure API Key and Endpoint
+Configure your Azure OpenAI API key and endpoint for this project, using .NET User Secrets:
+   1.  In the Azure Portal, navigate to your Azure OpenAI resource.
+   2.  Copy the "Endpoint" URL and "Key 1" from the "Keys & Endpoint" section.
+#### ---#if (hostIdentifier == "vs")
+   3. In Visual Studio, right-click on your project in the Solution Explorer and select "Manage User Secrets".
+   4. This will open a secrets.json file where you can store your API key and endpoint securely, add the following keys & values to the file:
```

Comment:
```suggestion
   4. This will open a secrets.json file where you can store your API key and endpoint without it being tracked in source control. Add the following keys & values to the file:
```